test.js

```diff
@@ -0,0 +1,73 @@
+var tests = {
+
+    	numbers: [1, 5, 66, 224, 6, 22, 4, 1, 6, 8, 9, 86, 3, 4, 32, 64, 10],
+
+    	/*
+	Given an integer array of size N, compute the sum of all even numbers in this array
+	*/
+	firstTest: function () {
+    		var array = this.numbers.slice();
+		//join the elements from array into a string
+		array = array.join();	    	
+		//searches the even numbers from string and returns the matches as an array
+		array = array.match(/\w?(\d+)*[24680]/g);
+	
+		//sum the elements from array by removing them
+		var result = 0;
+		for (var i = array.length; i--;) {
+		  	result += parseInt(array.pop());
+		}
+	    	return result;
+	},
+
+    	/*
+	Given an integer array of size N, identify the largest 2 numbers in this array. Do not sort the array.
+	*/
+	secondTest: function () { 
+	 	var array = this.numbers.slice();
+	    	//get the biggest number
+		var biggestNr = Math.max.apply(Math, array);
+			
+		//remove the biggest number from original array
+		var index = array.indexOf(biggestNr);
+		array.splice(index, 1);
+	
+		//get the second biggest number
+		var secondNr = Math.max.apply(Math, array);
+		
+		return biggestNr + ' & ' + secondNr;
+	},
+
+    	/*
+	Given an array of numbers having size N in which every number is between 1 and N, determine 
+	if there are any duplicates in it. You are allowed to destroy the array if you like. 
+	Do not sort the array or use bit vectors. Try to work within the array without using any other 
+	temporary data structures. Do not use 2 nesting for loops since the time complexity would be high. 
+	Try to find a solution by executing just one loop. Hint: manipulate the elements of the same array
+	as you loop through.
+	*/
+	thirdTest: function () {
+	    	var array = this.numbers.slice();
+		var tempValue = 0;
+		var result = '';
+	
+		for (var i = 0; i < array.length; i++) {
+			//store element in a temp variable
+			tempValue = array[i];
+	
+			//remove element from original array
+			var index = array.indexOf(array[i]);
```

Comment:
While I can't say this with 100% confidence for all Javascript implementations, internally, indexOf uses a separate loop which is forbidden by the problem requirements. e.g.: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf